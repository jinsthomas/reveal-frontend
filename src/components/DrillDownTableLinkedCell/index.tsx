import { DropDownCellProps } from '@onaio/drill-down-table';
import React from 'react';
import { Link } from 'react-router-dom';
import { FOCUS_INVESTIGATION_URL } from '../../constants';

/** Component that will be rendered in drop-down table cells showing a link
 * that moves you to the next hierarchical level.
 */
const DrillDownTableLinkedCell: React.ElementType<DropDownCellProps> = (
  props: DropDownCellProps
) => {
  const { cellValue, hasChildren } = props;
  return (
    <div>
      {hasChildren ? (
        <Link to={`${FOCUS_INVESTIGATION_URL}/${cellValue}`}>{cellValue}</Link>
      ) : (
        cellValue
      )}
    </div>
  );
};

export default DrillDownTableLinkedCell;
