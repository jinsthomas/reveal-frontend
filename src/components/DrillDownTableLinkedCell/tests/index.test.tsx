import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import { CellInfo } from 'react-table';
import DrillDownTableLinkedCell from '..';

const urlPath: string = 'http://example.com/path';
/* tslint:disable:object-literal-sort-keys */

const cell: CellInfo = {
  index: 0,
  viewIndex: 0,
  pageSize: 20,
  page: 0,
  level: 0,
  nestingPath: [0],
  column: {
    Header: '',
    show: true,
    minWidth: 90,
    className: '',
    style: {},
    headerClassName: '',
    headerStyle: {},
    footerClassName: '',
    footerStyle: {},
    filterAll: false,
    id: 'id',
  },
  value: '2019-11-20',
  resized: [],
  show: true,
  width: 90,
  tdProps: {
    rest: {},
  },
  columnProps: {
    rest: {},
  },
  classes: [],
  styles: {},
  pivoted: false,
  expander: false,
  isExpanded: false,
  maxWidth: 4,
  row: {},
  rowValues: {},
  aggregated: false,
  groupedByPivot: false,
  subRows: [],
  original: {},
};
/* tslint:disable:object-literal-sort-keys */

/**
 * Investiate why cellValue = 'some value' fails but
 * cellValue = cell.value works. Also why should either work if the component expects the value to be
 * of type Node?
 */
const cellValue = cell.value;
const history = createBrowserHistory();

describe('/components/DrillDownTableLinkedCell', () => {
  const propsHasChildren = {
    cell,
    cellValue,
    hasChildren: true,
    urlPath,
  };
  it('renders correctly if prop hasChildren is true', () => {
    const wrapper = mount(
      <Router history={history}>
        <DrillDownTableLinkedCell {...propsHasChildren} />
      </Router>
    );
    expect(wrapper.find(DrillDownTableLinkedCell).props()).toEqual(propsHasChildren);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders correctly if prop hasChildren is false', () => {
    const propsHasChildrenFalse = {
      ...propsHasChildren,
      hasChildren: false,
    };
    const wrapper = mount(
      <Router history={history}>
        <DrillDownTableLinkedCell {...propsHasChildrenFalse} />
      </Router>
    );
    expect(wrapper.find(DrillDownTableLinkedCell).props()).toEqual(propsHasChildrenFalse);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
