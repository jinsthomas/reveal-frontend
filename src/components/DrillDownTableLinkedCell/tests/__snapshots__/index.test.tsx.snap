// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/DrillDownTableLinkedCell renders correctly if prop hasChildren is false 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <DrillDownTableLinkedCell
    cell={
      Object {
        "aggregated": false,
        "classes": Array [],
        "column": Object {
          "Header": "",
          "className": "",
          "filterAll": false,
          "footerClassName": "",
          "footerStyle": Object {},
          "headerClassName": "",
          "headerStyle": Object {},
          "id": "id",
          "minWidth": 90,
          "show": true,
          "style": Object {},
        },
        "columnProps": Object {
          "rest": Object {},
        },
        "expander": false,
        "groupedByPivot": false,
        "index": 0,
        "isExpanded": false,
        "level": 0,
        "maxWidth": 4,
        "nestingPath": Array [
          0,
        ],
        "original": Object {},
        "page": 0,
        "pageSize": 20,
        "pivoted": false,
        "resized": Array [],
        "row": Object {},
        "rowValues": Object {},
        "show": true,
        "styles": Object {},
        "subRows": Array [],
        "tdProps": Object {
          "rest": Object {},
        },
        "value": "2019-11-20",
        "viewIndex": 0,
        "width": 90,
      }
    }
    cellValue="2019-11-20"
    hasChildren={false}
    urlPath="http://example.com/path"
  >
    <div>
      2019-11-20
    </div>
  </DrillDownTableLinkedCell>
</Router>
`;

exports[`/components/DrillDownTableLinkedCell renders correctly if prop hasChildren is true 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <DrillDownTableLinkedCell
    cell={
      Object {
        "aggregated": false,
        "classes": Array [],
        "column": Object {
          "Header": "",
          "className": "",
          "filterAll": false,
          "footerClassName": "",
          "footerStyle": Object {},
          "headerClassName": "",
          "headerStyle": Object {},
          "id": "id",
          "minWidth": 90,
          "show": true,
          "style": Object {},
        },
        "columnProps": Object {
          "rest": Object {},
        },
        "expander": false,
        "groupedByPivot": false,
        "index": 0,
        "isExpanded": false,
        "level": 0,
        "maxWidth": 4,
        "nestingPath": Array [
          0,
        ],
        "original": Object {},
        "page": 0,
        "pageSize": 20,
        "pivoted": false,
        "resized": Array [],
        "row": Object {},
        "rowValues": Object {},
        "show": true,
        "styles": Object {},
        "subRows": Array [],
        "tdProps": Object {
          "rest": Object {},
        },
        "value": "2019-11-20",
        "viewIndex": 0,
        "width": 90,
      }
    }
    cellValue="2019-11-20"
    hasChildren={true}
    urlPath="http://example.com/path"
  >
    <div>
      <Link
        replace={false}
        to="http://example.com/path/2019-11-20"
      >
        <a
          href="/http://example.com/path/2019-11-20"
          onClick={[Function]}
        >
          2019-11-20
        </a>
      </Link>
    </div>
  </DrillDownTableLinkedCell>
</Router>
`;
